crates/revrt/src/dataset.rs

```diff
@@ -78,39 +78,25 @@ impl Dataset {
             .map(|entry| entry.to_string())
             .find(|entry| {
                 let name = entry.split('/').next().unwrap_or("").to_ascii_lowercase();
+                // Skip coordinate axes when selecting a representative variable for cost storage.
                 const EXCLUDES: [&str; 6] =
                     ["latitude", "longitude", "band", "x", "y", "spatial_ref"];
                 !name.ends_with(".json") && !EXCLUDES.iter().any(|needle| name.contains(needle))
             })
             .expect("no suitable variables found in source dataset");
-        // Skip coordinate axes when selecting a representative variable for cost storage.
         let varname = first_entry.split('/').next().unwrap().to_string();
         debug!("Using '{}' to determine shape of cost data", varname);
         let tmp = zarrs::array::Array::open(source.clone(), &format!("/{varname}")).unwrap();
-        // let cost_shape = tmp.shape();
-        let chunk_shape = tmp.chunk_grid().clone();
-        debug!("Chunk grid info: {:?}", &chunk_shape);
-        // ----
-
-        trace!("Creating an empty cost array");
-        let array = zarrs::array::ArrayBuilder::new_with_chunk_grid(
-            // cost_shape,
-            chunk_shape,
-            zarrs::array::DataType::Float32,
-            zarrs::array::FillValue::from(zarrs::array::ZARR_NAN_F32),
-        )
-        .build(swap.clone(), "/cost")
-        .unwrap();
-        trace!("Cost shape: {:?}", array.shape().to_vec());
-        trace!("Cost chunk shape: {:?}", array.chunk_grid());
-        array.store_metadata().unwrap();
+        let chunk_grid = tmp.chunk_grid();
+        debug!("Chunk grid info: {:?}", &chunk_grid);
 
-        trace!("Cost dataset contents: {:?}", swap.list().unwrap());
+        add_layer_to_data("cost_invariant", chunk_grid, &swap);
```

Comment:
Not for this PR, but the concept of the delta cost layer could combine everything.